### PR TITLE
Added documentation

### PR DESCRIPTION
**Pull request by naved001**
What have I added here?

* Installation instructions (copied from the MOC wiki)
* Developer guidelines (based on HIL documentations)

### CODE REVIEW

File: docs/BMI-Installation.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
+# BMI
+
+The Bare Metal Imaging (BMI) is a core component of the Massachusetts Open Cloud and a image management system(ims) that  
+* provisions numerous nodes as quickly as possible while preserving support for multitenancy using Hardware as a Service (HaaS) and
```
Comment (chemistry-sourabh) [inline docs/BMI-Installation.md]: Change it to HIL. Everywhere

File: docs/BMI-Installation.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+```
+$ sudo python setup.py install
+$ pip install python-cephlibs
+```
+
+That's it. Installation is done!
+***
+## Running
+
+### Creating Tunnel (Only for internal purposes)
```
Comment (chemistry-sourabh) [inline docs/BMI-Installation.md]: Remove this from here. Not supposed to be in public repo

File: docs/BMI-Installation.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+```
+The general default port we use is 6500.  
+For PRB we made 2 tunnels
+
+* HIL's 127.0.0.1:80 -> moc02's 6500
+* moc02's 6500 -> BMI VM's 6500
+
+For Northeastern and Engage1 just the regular tunnel.
+
+### Configuration
```
Comment (chemistry-sourabh) [inline docs/BMI-Installation.md]: Will need to update this entire Section after PR merge

File: docs/BMI-Installation.md [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
+The Bare Metal Imaging (BMI) is a core component of the Massachusetts Open Cloud and a image management system(ims) that  
+* provisions numerous nodes as quickly as possible while preserving support for multitenancy using Hardware as a Service (HaaS) and  
+* introduces the image management techniques that are supported by virtual machines, with little to no impact on application performance.  
+
+***
+## Installation
+
+This document describes how to setup BMI
+
```
Comment (chemistry-sourabh) [inline docs/BMI-Installation.md]: I advise adding @sirushtim's script as the first way to install for developers. Then the manual installation can continue

**Comment by naved001** [comment]
@chemistry-sourabh Thanks! I'll make these changes

File: docs/BMI-Installation.md (hunk 2 above)
Comment (apoorvemohan) [inline docs/BMI-Installation.md]: @chemistry-sourabh what do you mean by "this"?

Comment (apoorvemohan) [inline docs/BMI-Installation.md]: Section "#### iSCSI Server" needs to be edited with correct way to setup TGT

Comment (chemistry-sourabh) [inline docs/BMI-Installation.md]: Creating the tunnel section

**Comment by chemistry-sourabh** [comment]
@naved001 are you waiting on something related to this PR?

File: docs/BMI-Installation.md (hunk 1 above)
Comment (naved001) [inline docs/BMI-Installation.md]: Done

File: docs/BMI-Installation.md (hunk 2 above)
Comment (naved001) [inline docs/BMI-Installation.md]: Done

File: docs/BMI-Installation.md (hunk 3 above)
Comment (naved001) [inline docs/BMI-Installation.md]: yes, will do that in the next commit.

File: docs/BMI-Installation.md (hunk 4 above)
Comment (naved001) [inline docs/BMI-Installation.md]: I think it would be better for sirushti to update the documents after his PR is merged.

**Comment by naved001** [comment]
@chemistry-sourabh @apoorvemohan updated the docs. have a look at it now.

File: docs/BMI-Installation.md (hunk 4 above)
Comment (chemistry-sourabh) [inline docs/BMI-Installation.md]: @sirushtim please check the installation process

**Comment by chemistry-sourabh** [comment]
Still didn't read the installation part. Will ping you when done.

File: docs/BMI-Installation.md [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+systemctl start tgtd
+chkconfig tgtd on
+systemctl status tgtd
+```
+
+For more information visit their [website](http://stgt.sourceforge.net/) and the [Quick Start Guide](https://fedoraproject.org/wiki/Scsi-target-utils_Quickstart_Guide)
+
+#### DHCP server
+You can setup `dnsmasq` for this.
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: This is mandatory and the only one we support as of now.

File: docs/BMI-Installation.md [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+`$ sudo apt-get install dnsmasq`
+
+* On CentOS:
+`$ yum install dnsmasq`
+
+For more information about dnsmasq, you can look at the [wiki](https://wiki.debian.org/HowTo/dnsmasq)
+
+#### TFTP Server
+##### On ubuntu:
+* Install these packages:  `$ sudo apt-get install xinetd tftpd tftp`
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: Why do we need a explicit TFTP, dnsmasq is going to bring that in. Unless, we go for another dnsserver, this section should be optional.

**Comment by chemistry-sourabh** [comment]
@ravisantoshgudimetla you also better hit the request changes button

File: docs/BMI-Installation.md [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+* verbose prints logs to screen
+```
+# this section is for logs
+[logs]
+path = /home/bmi/logs/
+debug = False
+verbose = False
+```
+
+### Einstein
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: We need to explain what Einstein is, if not just a reference here.

File: docs/BMI-Installation.md [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+```
+$ screen
+$ einstein_server (In screen)
+```
+
+Should see some lines stating einstein is running then hit Ctrl+A+D.  
+
+Einstein is running!!
+
+### Picasso
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: Same applies for picasso as well.

File: docs/Developer-guidlines.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+If fixing a bug, a regression test should accompany the bug fix to ensure that
+the bug does not return.
+
+### Pull Requests
+
+Once the checklist above has been met, open a Pull Request from your
+personal fork to the main BMI repo.
+
+
```
Comment (ravisantoshgudimetla) [inline docs/Developer-guidlines.md]: Add a reference to git pull request(Things like forking and raise a PR from branch etc).

Comment (chemistry-sourabh) [inline docs/Developer-guidlines.md]: you can add the link to the atlassian workflow tutorial.

File: docs/BMI-Installation.md (hunk 6 above)
Comment (naved001) [inline docs/BMI-Installation.md]: Ah, makes sense. I have made this section optional. I am fine with removing it too.

File: docs/BMI-Installation.md (hunk 7 above)
Comment (naved001) [inline docs/BMI-Installation.md]: I have put one line to tell what is Einstein and Picasso.

File: docs/Developer-guidlines.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+This summarizes what should be done prior to a pull request:
+
+- [ ] If functionality could have architectural implications or controversial, have a discussion with the team. Ideally, prior to coding to save effort.
+- [ ] Ensure any user, deployer or developer documentation is updated.
+- [ ] If a change affects an external API, be sure to update docs/rest\_api.md.
+- [ ] Testing:
+  - [ ] Ensure tests pass (coming soon!) after making your changes by running `py.test tests/unit ` from the top-level ims directory. Parallel testing can be used on multi-core systems by running `py.test tests/unit -n auto`
+  - [ ] Add unit tests in the corresponding file and create one if none are present.
+  - [ ] If practical, bug fixes should have an reproducing test to ensure that the bug does not come back.
+  - [ ] Run deployment tests if needed
```
Comment (naved001) [inline docs/Developer-guidlines.md]: > What do you mean by deployment tests ? Do you mean on the actual hardware ? If so they will run as part of the checks raised by the PR.

Yes, on actual hardware. But I don't understand what do you mean when you say "will run as part of the checks raised by the PR"

> So removing here and adding there would be better

Makes sense. Ill remove it from here and add it to a different doc.

File: docs/Developer-guidlines.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+#### Get agreement
```
Comment (naved001) [inline docs/Developer-guidlines.md]: I'll change it to "Get agreeement from the BMI team" (or if you have any other suggestion). Point being, before a developer starts coding, the team should, at least, agree with the idea to save coding effort.

File: docs/Developer-guidlines.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+### Code review
+
+Code reviews help increase code quality by finding:
+* mistakes that can oftentimes be overlooked by a single developer
+* improving readability
+* helping reviewers to learn about different areas of the code
+
+Reviewers are the final guardians for good code quality. If you see a piece of
+complex code, that is probably where you want to spend a lot of your review
+time. Remember the 80/20 rule (80% of the bugs come from 20% of the code).
```
Comment (naved001) [inline docs/Developer-guidlines.md]: what checks? :S

File: docs/Developer-guidlines.md (hunk 9 above)
Comment (naved001) [inline docs/Developer-guidlines.md]: I have linked the git workflow in Submitting code/Pull Requests section

File: docs/Developer-guidlines.md [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+* [Testing document](testing.html)
+
+Testing helps to ensure the quality of the code base. Every pull request
+submitted should first be tested to ensure that all existing tests pass.
+
+When introducing new functionality, new tests (both unit and more comprehensive)
+should be added that provide adequate coverage.
+
+If fixing a bug, a regression test should accompany the bug fix to ensure that
+the bug does not return.
```
Comment (naved001) [inline docs/Developer-guidlines.md]: Actually, this is pretty general stuff, I think we should let this be here. And the more detailed testing stuff (like how to run, and the test  structures etc) would go into the test docs.

File: docs/Developer-guidlines.md (hunk 12 above)
Comment (chemistry-sourabh) [inline docs/Developer-guidlines.md]: The pep8 and test cases that will run in openshift or travis

File: docs/Developer-guidlines.md (hunk 11 above)
Comment (chemistry-sourabh) [inline docs/Developer-guidlines.md]: What you suggested is fine.

**Comment by naved001** [comment]
@ravisantoshgudimetla @chemistry-sourabh @apoorvemohan I addressed your comments for this PR. Have a look at it, and if you are happy just squash and merge.

File: docs/BMI-Installation.md [hunk 14]
```diff
@@ -0,0 +1,287 @@
...
+* Run setup.py and install python-cephlibs
+```
+$ sudo python setup.py install
+$ pip install python-cephlibs
+```
+
+That's it. Installation is done!
+***
+## Running
+
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: What does running mean here?

File: docs/BMI-Installation.md [hunk 15]
```diff
@@ -0,0 +1,287 @@
...
+# This section is filesystem related config
+[fs]
+id = bmi
+pool = bmi
+conf_file = /etc/ceph/ceph.conf
+keyring = /etc/ceph/bmi.key
+```
+
+**Driver section**
+* net_isolator is the name of the network isolator to load
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: We should define, what network isolator is or atleast point to HIL documentation.

File: docs/BMI-Installation.md [hunk 16]
```diff
@@ -0,0 +1,287 @@
...
+[driver]
+fs = ims.einstein.fs.ceph.driver
+net_isolator = not used
+iscsi = ims.einstein.tgt.driver
+```  
+
+**iscsi section**
+* ip is the ip of iscsi server on the provisioning network
+* password is the sudo password for the VM (Will be removed)
+```
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: Add a notification here saying it will be deprecated in future.

File: docs/BMI-Installation.md [hunk 17]
```diff
@@ -0,0 +1,287 @@
...
+debug = False
+verbose = False
+```
+
+### Einstein  
+
+Einstein is the backend that does the operations. 
+
+First export the variable BMI_CONFIG
+
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: First, export the path of BMI config file to environment variable(BMI_CONFIG).

File: docs/Developer-guidlines.md [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+
+Reviewers are the final guardians for good code quality. If you see a piece of
+complex code, that is probably where you want to spend a lot of your review
+time. Remember the 80/20 rule (80% of the bugs come from 20% of the code).
+
+
+### Approval criteria
+
+Pull Requests require at least 2 successful reviews, also known as "+1"'s, in
+order to be merged, with at least 2 core developers being involved in some way:
```
Comment (ravisantoshgudimetla) [inline docs/Developer-guidlines.md]: Or approve in github.

File: docs/BMI-Installation.md (hunk 16 above)
Comment (chemistry-sourabh) [inline docs/BMI-Installation.md]: It is already deprecated if I remember right

File: docs/BMI-Installation.md (hunk 14 above)
Comment (naved001) [inline docs/BMI-Installation.md]: running bmi. I changed the title and moved this section where it makes more sense.

File: docs/BMI-Installation.md (hunk 16 above)
Comment (naved001) [inline docs/BMI-Installation.md]: It already says "will be removed". @chemistry-sourabh if it's already deprecated then I'll just remove this line.

File: docs/Developer-guidlines.md (hunk 18 above)
Comment (naved001) [inline docs/Developer-guidlines.md]: not sure what you mean here. It's understood that we are talking about approving and merging PRs in github.

File: docs/BMI-Installation.md (hunk 16 above)
Comment (chemistry-sourabh) [inline docs/BMI-Installation.md]: It is removed in TGT, but ceph still uses it for map and unmap which should be remove. So let it be.

File: docs/Developer-guidlines.md (hunk 18 above)
Comment (chemistry-sourabh) [inline docs/Developer-guidlines.md]: @ravisantoshgudimetla likes for everything to be explicit. Why not 2 approves instead of successful reviews ?

Comment (naved001) [inline docs/Developer-guidlines.md]: @chemistry-sourabh ah, good catch! when this was written there was no "approval" feature in github then. I'll update it.

File: docs/BMI-Installation.md [hunk 19]
```diff
@@ -0,0 +1,288 @@
...
+* password is the sudo password for the VM (Will be removed
+```
+# This section is for iscsi related config
+[iscsi]
+ip = 127.0.0.1
+password = password
+```  
+
+**rpc section**
+* name_server ip and port is the RPC NameServer's IP and Port
```
Comment (chemistry-sourabh) [inline docs/BMI-Installation.md]: is the ip and port Name Server binds to ( Similar to RPC server also)

File: docs/Developer-guidlines.md (hunk 13 above)
Comment (chemistry-sourabh) [inline docs/Developer-guidlines.md]: In that case Add the link at the ending of the section saying refer for more details

File: docs/Developer-guidlines.md [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+the bug does not return.
+
+Once the checklist above has been met, open a Pull Request from your
+personal fork to the main BMI repo.
+
+
+### Code review
+
+Code reviews help increase code quality by finding:
+* mistakes that can oftentimes be overlooked by a single developer
```
Comment (chemistry-sourabh) [inline docs/Developer-guidlines.md]: often times

File: docs/BMI-Installation.md [hunk 21]
```diff
@@ -0,0 +1,288 @@
...
+### Prerequisite Softwares
+
+BMI requires the following non python dependencies. 
+
+#### Ceph Client
+Ceph is a distributed object store and file system designed for performance and scalibility. 
+For more information, visit the [official website](http://docs.ceph.com/docs/master/start/)
+
+You can follow [this guide](http://palmerville.github.io/2016/04/30/single-node-ceph-install.html) 
+to setup a ceph client on a VM
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: It seems the link points to single node ceph installation. What we need is a client.(and it will vary from from one OS to other). So, we should be explicit in mentioning it.(the link points to ubuntu and can work for all debian flavours, I guess).

File: docs/BMI-Installation.md [hunk 22]
```diff
@@ -0,0 +1,288 @@
...
+make CEPH_RBD=1
+make CEPH_RBD=1 install
+
+
+cp scripts/tgtd.service /usr/lib/systemd/system/
+systemctl daemon-reload
+systemctl list-unit-files | grep -i tgt
+
+iptables -I INPUT -p tcp -m tcp --dport 3260 -j ACCEPT
+service iptables save
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: Again, we should talk about iptables, only if OS is RHEL family.

**Comment by ravisantoshgudimetla** [comment]
I guess, OS related things need to be explicitly mentioned like "On RHEL family or on ubuntu etc", so that there are no ambiguities while installation. I think if we mention that as of now we are supporting RHEL family makes much sense. I will make one more pass in coming week.

File: docs/Developer-guidlines.md (hunk 18 above)
Comment (ravisantoshgudimetla) [inline docs/Developer-guidlines.md]: > Or approve in github.

Thats what I meant earlier, I should have been explicit :).

File: docs/Developer-guidlines.md (hunk 20 above)
Comment (naved001) [inline docs/Developer-guidlines.md]: this is so meta lol.

File: docs/Developer-guidlines.md (hunk 13 above)
Comment (naved001) [inline docs/Developer-guidlines.md]: added a link to `testing.md`

File: docs/Developer-guidlines.md (hunk 12 above)
Comment (naved001) [inline docs/Developer-guidlines.md]: okay, updated that in the "Prior to the pull request" section. Line 52 and 52

File: docs/Developer-guidlines.md (hunk 20 above)
Comment (chemistry-sourabh) [inline docs/Developer-guidlines.md]: lol googled it, looks like oftentimes is an actual word. English is truly a weird language.

File: docs/BMI-Installation.md [hunk 23]
```diff
@@ -0,0 +1,288 @@
...
+
+* On ubuntu:
+`$ sudo apt-get install dnsmasq`
+
+* On CentOS:
+`$ yum install dnsmasq`
+
+For more information about dnsmasq, you can look at the [wiki](https://wiki.debian.org/HowTo/dnsmasq)
+
+#### TFTP Server (Optional)
```
Comment (naved001) [inline docs/BMI-Installation.md]: cool, got rid of it.

File: docs/BMI-Installation.md (hunk 21 above)
Comment (naved001) [inline docs/BMI-Installation.md]: good catch. I update this section.

File: docs/Developer-guidlines.md (hunk 20 above)
Comment (naved001) [inline docs/Developer-guidlines.md]: actually, oftentimes is a valid word

**Comment by naved001** [comment]
okay. changed some more stuff around here. let me know if I missed out on something.

File: docs/BMI-Installation.md [hunk 24]
```diff
@@ -0,0 +1,240 @@
...
+For more information, visit the [official website](http://docs.ceph.com/docs/master/start/)
+
+To install ceph client on RHEL:
+```
+yum install ceph-common
+```
+
+To install ceph client on ubuntu:
+```
+apt-get install ceph-common
```
Comment (chemistry-sourabh) [inline docs/BMI-Installation.md]: Remove the ubuntu related commands. We said we officially support RHEL, so no need for this. Also iSCSI section doesnt have an ubuntu section so better remove it.

File: docs/BMI-Installation.md [hunk 25]
```diff
@@ -0,0 +1,240 @@
...
+systemctl status tgtd
+```
+
+For more information visit their [website](http://stgt.sourceforge.net/) and the [Quick Start Guide](https://fedoraproject.org/wiki/Scsi-target-utils_Quickstart_Guide)
+
+#### DHCP server
+We support `dnsmasq` for this. 
+
+* On ubuntu:
+`$ sudo apt-get install dnsmasq`
```
Comment (chemistry-sourabh) [inline docs/BMI-Installation.md]: Same like above

File: docs/Developer-guidlines.md [hunk 26]
```diff
@@ -0,0 +1,109 @@
...
+discussed several alternatives to how things currently work (which we're trying
+to get better about writing down), and it would be good to have agreement that
+includes input from these past discussions as well as the wisdom of the
+community. The best way to do this is to [file an
+issue](https://github.com/CCI-MOC/ims/issues) on github, or speak with one of the core developers directly.
+
+
+#### Testing
+
+* [Testing document](testing.md)
```
Comment (chemistry-sourabh) [inline docs/Developer-guidlines.md]: Remove this

File: docs/Developer-guidlines.md [hunk 27]
```diff
@@ -0,0 +1,109 @@
...
+
+### Approval criteria on github
+
+Pull Requests require at least 2 approvals, also known as "+1"'s, in
+order to be merged, with at least 2 core developers being involved in some way:
+either as a submitter or a reviewer.
+
+The exception to this is documentation, where we are a little more lenient in
+the interest of lowering the barrier to having better docs. For docs changes,
+typically one +1 is sufficient.
```
Comment (chemistry-sourabh) [inline docs/Developer-guidlines.md]: We are not following this para so should it be there ?

File: docs/Developer-guidlines.md [hunk 28]
```diff
@@ -0,0 +1,109 @@
...
+
+Whomever provides the enabling +1 is responsible for clicking the merge button
+on github. If you do not have commit access, then please add your +1 to the PR
+and ask one of the Core Developers to complete the merge.
```
Comment (chemistry-sourabh) [inline docs/Developer-guidlines.md]: Even this is not followed. We are asking whoever is the team lead before merging.

File: docs/Developer-guidlines.md (hunk 27 above)
Comment (naved001) [inline docs/Developer-guidlines.md]: I think that is because we are in the early stages. Once we are done with 0.5 issues, we'll be following these.

File: docs/Developer-guidlines.md (hunk 28 above)
Comment (naved001) [inline docs/Developer-guidlines.md]: ^^same as above. Right now we are a bit more paranoid which is why we are asking the team lead and others for opinions on most PRs.

File: docs/BMI-Installation.md (hunk 24 above)
Comment (naved001) [inline docs/BMI-Installation.md]: cool. will get rid of ubuntu stuff.

**Comment by chemistry-sourabh** [comment]
ok then

**Comment by chemistry-sourabh** [comment]
Go have a Beer!!

File: docs/BMI-Installation.md [hunk 29]
```diff
@@ -0,0 +1,231 @@
...
+That's it. Installation is done!
+***
+
+## Configuration
+
+The template for the config is [here](https://github.com/CCI-MOC/ims/blob/dev/bmi_config.cfg)
+
+This section describes each config section along with sample  
+
+**bmi section**
```
Comment (sirushtim) [inline docs/BMI-Installation.md]: This section can get easily outdated. Filed a bug for this: https://github.com/CCI-MOC/ims/issues/105

Comment (chemistry-sourabh) [inline docs/BMI-Installation.md]: Ya probably we can separate it out later when the issue is addressed

**Comment by chemistry-sourabh** [comment]
@ravisantoshgudimetla your approve is left

File: docs/BMI-Installation.md [hunk 30]
```diff
@@ -0,0 +1,231 @@
...
+yum install -y gcc
+** delete install doc from Makefile if problem occurs in the next three steps
+make CEPH_RBD=1 clean
+make CEPH_RBD=1
+make CEPH_RBD=1 install
+
+
+cp scripts/tgtd.service /usr/lib/systemd/system/
+systemctl daemon-reload
+systemctl list-unit-files | grep -i tgt
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: We can tell that this command is for checking if tgtd has been made as a deamon

File: docs/BMI-Installation.md (hunk 29 above)
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: Probably a link to sample file in git(which will be updated periodically is good enough).

File: docs/BMI-Installation.md [hunk 31]
```diff
@@ -0,0 +1,231 @@
...
+This section describes each config section along with sample  
+
+**bmi section**  
+* uid is used to prevent clashing of BMIs that use the same Ceph Pool. It is the responsibility of the admin to ensure that uid is unique.
+* service should be set to True if BMI is deployed as a service. Set it to False for now as there are some glitches.
+```
+# this section is for basic bmi settings
+[bmi]
+# uid is given so that images dont clash in ceph pool
+uid = 5
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: Do, we need to explain it twice. It seems you already did a good job explaining what it is and why it is needed, either move those to comments to lines above the parameters or remove the comment line here.

Comment (chemistry-sourabh) [inline docs/BMI-Installation.md]: or as you suggested the enhanced explanations can be moved to the sample config file

File: docs/BMI-Installation.md [hunk 32]
```diff
@@ -0,0 +1,231 @@
...
+# This section is filesystem related config
+[fs]
+id = bmi
+pool = bmi
+conf_file = /etc/ceph/ceph.conf
+keyring = /etc/ceph/bmi.key
+```
+
+**Driver section**
+* net_isolator is the name of the network isolator to use. it connects the
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: The main idea is to have multi-tenancy while provisioning(securely).

File: docs/BMI-Installation.md [hunk 33]
```diff
@@ -0,0 +1,231 @@
...
+* node to a network on which BMI can provision. We use HIL for network
+* isolation. [Link to HIL](http://hil.readthedocs.io/en/latest/)
+* iscsi is the iscsi driver to load
+* fs is the filesystem to use
+```
+# This section is for hil related config
+[driver]
+fs = ims.einstein.fs.ceph.driver
+net_isolator = not used
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: <not used> may sound odd. Can we change this to NULL or something much more meaningful.

File: docs/BMI-Installation.md [hunk 34]
```diff
@@ -0,0 +1,231 @@
...
+# This section is for iscsi related config
+[iscsi]
+ip = 127.0.0.1
+password = password
+```  
+
+**rpc section**
+* name_server is the ip and port Name Server binds to
+* rpc_server is the ip and port RPC Server binds to (The end which calls einstein)
+```
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: We should talk about pyro or atleast have a link to it.(https://pypi.python.org/pypi/Pyro4)

File: docs/BMI-Installation.md [hunk 35]
```diff
@@ -0,0 +1,231 @@
...
+* pxelinux_path is the path to the pxelinux.cfg folder
+* ipxe_path is the path to the location where ipxe files should be created (root of tftpboot folder)
+```
+# this section is for specifying tftp settings
+[tftp]
+pxelinux_path = /var/lib/tftpboot/pxelinux.cfg
+ipxe_path = /var/lib/tftpboot/
+```  
+**http section**
+* bind_ip and port is the ip and port picasso should bind to
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: I am thinking out loud here. Should this be http section or API server section?

File: docs/BMI-Installation.md [hunk 36]
```diff
@@ -0,0 +1,231 @@
...
+path = /home/bmi/logs/
+debug = False
+verbose = False
+```
+## Running BMI
+
+### Einstein  
+
+Einstein is the backend that does the operations. 
+
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: I think, this should be somewhere in the starting, where we talked about Einstein first.

File: docs/BMI-Installation.md [hunk 37]
```diff
@@ -0,0 +1,231 @@
...
+* path is the folder where logs should be generated
+* debug enables debug logs
+* verbose prints logs to screen
+```
+# this section is for logs
+[logs]
+path = /home/bmi/logs/
+debug = False
+verbose = False
+```
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: Do we have some default values for every parameter, if yes, we should talk about the default values as well.

File: docs/BMI-Installation.md [hunk 38]
```diff
@@ -0,0 +1,231 @@
...
+$ screen
+$ picasso_server (In screen)
+```
+
+Should see one line stating that Picasso is running  
+
+Picasso is running!!
+
+### Initing DB
+
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: Bootstrapping DB, may be a better term, nit again, could be ignored.

File: docs/BMI-Installation.md (hunk 29 above)
Comment (naved001) [inline docs/BMI-Installation.md]: there's already a link to a sample config file with every section explained in it (it's more like a template).

File: docs/BMI-Installation.md (hunk 30 above)
Comment (naved001) [inline docs/BMI-Installation.md]: good idea!

File: docs/BMI-Installation.md (hunk 33 above)
Comment (naved001) [inline docs/BMI-Installation.md]: I have set this to <default> to indicate that it's using the default net isolator we have right now.

File: docs/BMI-Installation.md (hunk 37 above)
Comment (naved001) [inline docs/BMI-Installation.md]: 1 is a path and 2 of those are booleans, and those are self explanatory? aren't they? moreover, I am not sure if you can have a default value here, it's just for reference anyway.

File: docs/BMI-Installation.md (hunk 36 above)
Comment (naved001) [inline docs/BMI-Installation.md]: This is the only place where we talk about Einstein. In this installation document, we first talk about installing and configuring the Prerequisite softwares, then configuring the config file, and finally running the BMI services (Einstein and Picasso)

**Comment by naved001** [comment]
@ravisantoshgudimetla I have made some changes, let me know if this looks good.

File: docs/BMI-Installation.md [hunk 39]
```diff
@@ -0,0 +1,132 @@
+# BMI
+
+
+## Installation
+
+This document describes how to setup BMI. We officially support RHEL and it's
+variants, but it may also work on ubuntu (Debian) like OSes.
```
Comment (ravisantoshgudimetla) [inline docs/BMI-Installation.md]: NIT: OSes doesn't make sense. Probably expanding it is clear.
